src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
     )
 }
 
-function statFile(file: vscode.Uri) {
-    return statSync(file.fsPath).size
+interface BucketQuickPickItem extends vscode.QuickPickItem {
+    bucket: S3.Bucket | undefined
 }
 
-async function refreshNode(node: S3BucketNode | S3FolderNode, commands: Commands): Promise<void> {
-    node.clearChildren()
-    return commands.execute('aws.refreshAwsExplorerNode', node)
+/**
+ * 
+ * @returns 
+ */
+async function promptUserForBucket(): Promise<S3.Bucket | undefined> {
+    let regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1" //TODO:: decide how to get the regionCode!!
```

Comment:
Remove TODO unless you intend to reconsider this decision. Nit: space before `??`